Wrappers/Python/cil/utilities/callbacks.py

```diff
@@ -0,0 +1,85 @@
+
+from abc import ABC, abstractmethod
+
+from tqdm.auto import tqdm as tqdm_auto
+from tqdm.std import tqdm as tqdm_std
+
+
+class Callback(ABC):
+    '''Base Callback to inherit from for use in :code:`Algorithm.run(callbacks: list[Callback])`.'''
+    def __init__(self, verbose=1):
+        ''':param verbose: int, 0=quiet, 1=info, 2=debug'''
+        self.verbose = verbose
+
+    @abstractmethod
+    def __call__(self, algorithm):
+        pass
+
+
+class OldCallback(Callback):
+    '''Converts an old-style :code:`def callback(iteration, objective, x)`
+    to a new-style :code:`class Callback`.
+    '''
+    def __init__(self, callback, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.func = callback
+
+    def __call__(self, algorithm):
+        if algorithm.update_objective_interval > 0 and algorithm.iteration % algorithm.update_objective_interval == 0:
+            self.func(algorithm.iteration, algorithm.get_last_objective(return_all=self.verbose), algorithm.x)
+
+
+class ProgressCallback(Callback):
+    ''':code:`tqdm`-based progress bar.'''
```

Comment:
The list of parameters should go in the class description rather than in in the init